compiler/packages/babel-plugin-react-compiler/src/Inference/InferMutationAliasingEffects.ts

```diff
@@ -467,6 +475,7 @@ function applyEffect(
       break;
     }
     case 'CreateFunction': {
+      effects.push(effect);
```

Comment:
a consistent theme is that effects should almost always be preserved rather than dropped, so that range analysis can understand them. here, we want to know about the creation